File: app/src/main/java/com/raival/fileexplorer/utils/APKSignerUtils.java
package com.raival.fileexplorer.utils;

import static com.raival.fileexplorer.utils.D8Utils.unzipFromAssets;

import com.raival.fileexplorer.App;

import java.io.File;

public class APKSignerUtils {
    public static File getPk8() {
        File check = new File(App.appContext.getFilesDir() + "/build/testkey.pk8");
        if (check.exists()) {
            return check;
        }
        unzipFromAssets(App.appContext, "testkey.pk8.zip", check.getParentFile().getAbsolutePath());
        return check;
    }

    public static File getPem() {
        File check = new File(App.appContext.getFilesDir() + "/build/testkey.x509.pem");
        if (check.exists()) {
            return check;
        }
        unzipFromAssets(App.appContext, "testkey.x509.pem.zip", check.getParentFile().getAbsolutePath());
        return check;
    }
}

File: app/src/main/java/com/raival/fileexplorer/tab/file/misc/APKSignerUtils.kt
package com.raival.fileexplorer.tab.file.misc

import com.raival.fileexplorer.App
import com.raival.fileexplorer.tab.file.misc.BuildUtils.unzipFromAssets
import java.io.File

object APKSignerUtils {
    val pk8: File
        get() {
            val check = File(App.appContext.filesDir.toString() + "/build/testkey.pk8")
            if (check.exists()) {
                return check
            }
            unzipFromAssets(
                App.appContext,
                "build/testkey.pk8.zip",
                check.parentFile?.absolutePath
            )
            return check
        }
    val pem: File
        get() {
            val check = File(App.appContext.filesDir.toString() + "/build/testkey.x509.pem")
            if (check.exists()) {
                return check
            }
            unzipFromAssets(
                App.appContext,
                "build/testkey.x509.pem.zip",
                check.parentFile?.absolutePath
            )
            return check
        }
}